src/Symfony/Component/DependencyInjection/Exception/ContextElementNotFoundException.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Exception;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+class ContextElementNotFoundException extends InvalidArgumentException
+{
+    public function __construct($key, \Exception $previous = null)
```

Comment:
Introducing optional argument later is fine, but removing useless ones is a no-go for BC reasons. So I won't introduce something not really needed. Lazy is good for BC promise ;)

I'll even remove the `$exception` argument.